src/token.sw

```diff
@@ -0,0 +1,83 @@
+library token;
+//! Functionality for performing common operations on tokens.
+
+use ::address::Address;
+use ::chain::panic;
+
+// @todo if tx format changes, the magic number "48" must be changed !
+// TransactionScript outputsCount has a 48 byte(6 words * 8) offset
+// Transaction Script: https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#transactionscript
+const OUTPUT_LENGTH_LOCATION = 48;
+// Output types: https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#output
+const OUTPUT_VARIABLE_TYPE = 4;
+
+/// Mint `n` coins of the current contract's asset_id.
+pub fn mint(n: u64) {
+    asm(r1: n) {
+        mint r1;
+    }
+}
+
+/// Burn `n` coins of the current contract's asset_id.
+pub fn burn(n: u64) {
+    asm(r1: n) {
+        burn r1;
+    }
+}
+
+/// Transfer amount `coins` of type `asset_id` to address `recipient`.
+pub fn transfer_to_output(coins: u64, asset_id: b256, recipient: Address) {
```

Comment:
Hmmm. `coins` isn't really the most evocative. Would `amount` be better? Also, `asset_id`'s type should be changed to `ContractId` at some point, yes?

Reply:
I've changed this to be: "Transfer `amount` coins of type..."
I think it reads better.